### PR TITLE
Make --postfix-projections the default

### LINKED ISSUE #6943: Making `--exact-split` and `--postfix-projections` default?
I wonder when is a good time to change some defaults?
- [ ] 2.6.5?
- [x] or do we have to bump to 2.7 then?

Candidates for defaults:
- [x] `--exact-split`
- [x] `--postfix-projections`
- [ ] `--prop`

Anything else?

- [x] `--keep-pattern-variables`
- [ ] `--erasure`

I think it would be good to bundle such changes so migration (like adding `flags` to `.agda-lib` files) can be done in one go.

### LINKED ISSUE #7318: `--postfix-projections` do not make use of mixfix syntax
Consider the testcase for #2288:
```agda
record Setoid : Set₁ where
  field
    ∣_∣    : Set
    {{eq}} : Eq ∣_∣
open Setoid

NatSetoid : Setoid
∣ NatSetoid ∣ = Nat
```
The error with `--no-postfix-projections` is: 
```
No instance of type Eq ∣ NatSetoid ∣ ...
```
With `--postfix-projections` we get:
```
No instance of type Eq (NatSetoid .∣_∣)
```
This isn't pretty.  Postfix notation should not kick in when we can actually use the mixfix-syntax.

A little doubt: the `field ∣_∣ : Set` is actually a hack, exploiting the questionable "feature" of Agda to blindly apply mixfix syntax even when outside the module it was defined in.
- #632

So maybe the current broken postfix printing is less wrong than the prefix printing.

Opinions?

### PR DESCRIPTION
**Pull request by andreasabel**
Changing the default to `--postfix-projections` for 2.7.0 was discussed during one of our dev meetings.

Also closes #7318.
- #7318
- #6943

### CODE REVIEW

File: test/Fail/Issue1952.err [hunk 1]
```diff
@@ -99,75 +99,78 @@ Failed to solve the following constraints:
     D : Structure
     C : Structure
     (stuck)
-  (_r_173 Structure.⟶
-   (_r_173 Structure.× (_r_173 Structure.× _X_174) _X_174) _X_174)
+  _r_173 .Structure._⟶_
+  (_r_173 .Structure._×_ (_r_173 .Structure._×_ _X_174 _X_174)
+   _X_174)
```
Comment (andreasabel) [inline test/Fail/Issue1952.err]: Here the printing became accidentially better, but the underlying problem is #632.

File: test/Fail/Issue3606.err [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Failed to solve the following constraints:
-  fst ?0 = fst x : A (blocked on _5)
+  ?0 .fst = fst x : A (blocked on _5)
```
Comment (jespercockx) [inline test/Fail/Issue3606.err:2]: Why is the first `fst` here converted to postfix syntax but the second one not?

Comment (andreasabel) [reply test/Fail/Issue3606.err:2]: The user wrote a prefix projection in `fst x`:
```agda
test : (x : ΣAP) → P (fst x)
test x = snd {!!}
```
These are not touched by the `--postfix-projection` option.
If the user wants a consistent experience, they should use postfix projections now always or switch to `--no-postfix-projections`.

**Review by jespercockx — APPROVED** [review]

File: test/Fail/Issue3606.err (hunk 2 above)
Comment (jespercockx) [reply test/Fail/Issue3606.err:2]: Ok, I see.

**Review by plt-amy — APPROVED** [review]

**Pull request merged**